eclair-core/src/test/scala/fr/acinq/eclair/router/RouteCalculationSpec.scala

```diff
@@ -1884,6 +1884,28 @@ class RouteCalculationSpec extends AnyFunSuite with ParallelTestExecution {
       assert(findMultiPartRoute(g, a, b, amount, 101_000 msat, Set.empty, Set.empty, Set.empty, Nil, routeParams = routeParams, currentBlockHeight = BlockHeight(400000)).isSuccess)
     }
   }
+
+  test("small local edge with liquidity is better than big remote edge") {
+    // A === B === C -- D
+    //  \_________/
+    val g = DirectedGraph(List(
+      makeEdge(1L, a, b, 100 msat, 100, minHtlc = 1000 msat, capacity = 100000000 sat),
```

Comment:
This edge should also have its balance known to be representative of a real scenario since it's a local channel:
```suggestion
      makeEdge(1L, a, b, 100 msat, 100, minHtlc = 1000 msat, capacity = 100000000 sat, balance_opt = Some(150000 msat)),
```

I made it have an even bigger balance than the A -> C one to showcase that it's still ignored because the A -> C -> D route is better regardless (we only need balance to forward the payment, any "extra" balance doesn't matter for local channels).

Reply:
Done